monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/MapParallelOrderedObservable.scala

```diff
@@ -160,6 +157,7 @@ private[reactive] final class MapParallelOrderedObservable[A, B](
 
           case Failure(error) =>
             lastAck = Stop
+            composite.cancel()
```

Comment:
the order should be different, no gain in `composite -= future.cancelable` after it is canceled

Reply:
> the order should be different,
Do we even need this line now as we can just cancel the composite?

Reply:
I'm not 100% so I'd prefer to stay with status quo

Reply:
Hmm, Ok. Done